### PR TITLE
Blobstore: use new sha384 in blobstore

### PR DESCRIPTION
**Pull request by urosj**
- Did not remove the multihash package with this PR, if all is well, we can remove the files.

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/189/

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/190/

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/191/

File: internal/blobstore/blobstore.go [hunk 1]
```diff
@@ -52,7 +48,7 @@ type ContentChallengeResponse struct {
 
 // NewHash is used to calculate checksums for the blob store.
 func NewHash() hash.Hash {
-	return multihash.New(md5.New(), sha256.New())
+	return sha512.New384()
```
Comment (frankban) [inline internal/blobstore/blobstore.go:51]: Cool.

File: internal/blobstore/blobstore.go [hunk 2]
```diff
@@ -108,7 +103,7 @@ func (s *Store) challengeResponse(resp *ContentChallengeResponse) error {
 // satisfied by a client to prove that they have access to the content.
 // If the proof has already been acquired, it should be passed in as the
 // proof argument.
-func (s *Store) Put(r io.Reader, name string, size int64, hash string, proof *ContentChallengeResponse) (*ContentChallenge, error) {
+func (s *Store) Put(r io.Reader, name string, size int64, hash384 string, proof *ContentChallengeResponse) (*ContentChallenge, error) {
```
Comment (frankban) [inline internal/blobstore/blobstore.go]: FWIW hash was ok as a name: I don't see the need to specofy the underlying algorithm in the variable name.

File: internal/blobstore/blobstore.go [hunk 3]
```diff
@@ -121,11 +116,10 @@ func (s *Store) Put(r io.Reader, name string, size int64, hash string, proof *Co
 		// was created, so continue on with uploading
 		// the content as if there was no previous challenge.
 	}
-	rh := newResourceHash(hash)
-	resp, err := s.mstore.PutForEnvironmentRequest("", name, rh)
+	resp, err := s.mstore.PutForEnvironmentRequest("", name, hash384)
```
Comment (frankban) [inline internal/blobstore/blobstore.go]: Ditto.

File: internal/multihash/multihash.go [hunk 4]
```diff
@@ -1,44 +1,45 @@
 package multihash
```
Comment (frankban) [inline internal/multihash/multihash.go]: Why commenting this out and not just removing the file?

File: internal/multihash/multihash_test.go [hunk 5]
```diff
@@ -1,40 +1,40 @@
 package multihash_test
 
-import (
```
Comment (frankban) [inline internal/multihash/multihash_test.go:3]: I think we can just remove this file too.
We have VCS if we will need this again in the future.

**Comment by frankban** [comment]
:+1: with minor fixes.
Thank you!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/192/

**Pull request merged**